### PR TITLE
fluent operator & fluentbit: Added tolerations, nodeSelector + more

### LINKED ISSUE #695: Add ability to define tolerations and security contexts in the operator chart
### Is your feature request related to a problem? Please describe.

Right now I cannot deploy the operator - the pod stays in pending as it is missing tolerations. There is no way of defining this as it's not in the deployment yaml in the chart.

Same with security contexts.

These are required to deploy in an enterprise multi tenant environment.

### Describe the solution you'd like

Ability to define tolerations and security context in the main values.yaml.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by felfa01**
### What this PR does / why we need it:
Added support for tolerations, nodeSelector, priorityClassName and podSecurityContext to fluent-operator deployment and fluentbit daemonset

### Which issue(s) this PR fixes:

Fixes https://github.com/fluent/helm-charts/issues/314 
Fixes #695 

### Does this PR introduced a user-facing change?

```release-note
Added support for tolerations, nodeSelector, priorityClassName and podSecurityContext to fluent-operator deployment and fluentbit daemonset
```

### Additional documentation, usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by illrill** [comment]
@felfa01 I contributed a podSecurityContext to fully resolve https://github.com/fluent/fluent-operator/issues/695. Can you update the title and description to mention both fluent-bit and fluent-operator?

**Comment by felfa01** [comment]
> @felfa01 I contributed a podSecurityContext to fully resolve #695. Can you link it, and update the title and description to mention both fluent-bit and fluent-operator?

Thanks @illrill! It is updated.

File: charts/fluent-operator/templates/fluent-operator-deployment.yaml [hunk 1]
```diff
@@ -95,7 +95,18 @@ spec:
...
       serviceAccountName: fluent-operator
+      {{- if .Values.operator.priorityClassName }}
+      priorityClassName: {{ .Values.operator.priorityClassName | quote }}
+      {{- end }}
       {{- if .Values.operator.imagePullSecrets }}
-      imagePullSecrets:
-      {{- toYaml .Values.operator.imagePullSecrets | nindent 8 }}
+      imagePullSecrets: {{ toYaml .Values.operator.imagePullSecrets | nindent 8 }}
+      {{- end }}
+      {{- if .Values.operator.tolerations }}
```
Comment (wenchajun) [inline charts/fluent-operator/templates/fluent-operator-deployment.yaml:104]: maybe 
```
      imagePullSecrets: 
      {{ toYaml .Values.operator.imagePullSecrets | nindent 8 }}
      {{- end }}
```
better

File: charts/fluent-operator/templates/fluent-operator-deployment.yaml [hunk 2]
```diff
@@ -95,7 +95,18 @@ spec:
...
+      tolerations: {{ toYaml .Values.operator.tolerations | nindent 8 }}
```
Comment (wenchajun) [inline charts/fluent-operator/templates/fluent-operator-deployment.yaml:105]: ditto

File: charts/fluent-operator/templates/fluent-operator-deployment.yaml [hunk 3]
```diff
@@ -95,7 +95,18 @@ spec:
...
+      {{- end }}
+      {{- if .Values.operator.nodeSelector }}
+      nodeSelector: {{ toYaml .Values.operator.nodeSelector | nindent 8 }}
```
Comment (wenchajun) [inline charts/fluent-operator/templates/fluent-operator-deployment.yaml:108]: ditto

File: charts/fluent-operator/templates/fluent-operator-deployment.yaml [hunk 4]
```diff
@@ -95,7 +95,18 @@ spec:
...
+      {{- end }}
+      {{- if .Values.operator.podSecurityContext }}
+      podSecurityContext: {{ toYaml .Values.operator.podSecurityContext | nindent 8 }}
```
Comment (wenchajun) [inline charts/fluent-operator/templates/fluent-operator-deployment.yaml:111]: ditto

File: charts/fluent-operator/templates/fluent-operator-deployment.yaml (hunk 1 above)
Comment (illrill) [reply charts/fluent-operator/templates/fluent-operator-deployment.yaml]: Indentation style is now aligned with the rest of the manifest.

**Review by wenchajun — APPROVED** [review]

**Review by benjaminhuo — APPROVED** [review]

**Comment by benjaminhuo** [comment]
@felfa01 Thank you!

**Pull request merged**